update.sh

```diff
@@ -96,7 +96,7 @@ function in_versions_to_update() {
 function in_variants_to_update() {
   local variant=$1
 
-  if [ "${#update_variants[@]}" -eq 0 ]; then
+  if [ "${#variant_arg[@]}" -eq 0 ]; then
```

Comment:
This one I don't understand. Can you explain what this does?

Reply:
`update_variants` is an array which contains all variants given as arguments that exist as variants. If `update.sh` is invoked with unknown (non-existing) variant, `update_variants` is an empty array, and the condition here proceeds to update all variants. Changing the condition to compare against `variant_arg`, then all variants are updated in the case no arguments were passed in invocation instead of the case where unknown variant is passed as variant argument.

Before this change:
```
$ ./update.sh 8 foo
Updating version 8...
8/jessie-slim/Dockerfile updated!
8/jessie/Dockerfile updated!
8/onbuild/Dockerfile is already up to date!
8/alpine/Dockerfile updated!
8/stretch/Dockerfile updated!
8/stretch-slim/Dockerfile updated!
8/buster/Dockerfile updated!
8/buster-slim/Dockerfile updated!
Done!
```
After this change:
```
$ ./update.sh 8 foo
Updating version 8...
Done!
```